src/main/java/bz/turtle/readable/input/Feature.java

```diff
@@ -84,22 +111,38 @@ public String toString() {
    */
   public void rename(int name) {
     if (name < 0) {
-      rename("" + name);
+      this.setName(name);
```

Comment:
i think this should reset the hash, because previously it was reset in rename(string)
and now setName does not reset the computed hash

Reply:
Yes you are absolutely right! Good catch